app/addons/news/components.js

```diff
@@ -9,17 +9,73 @@
 // WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 // License for the specific language governing permissions and limitations under
 // the License.
+import app from "../../app";
 
 import React from "react";
 
-const NewsPage = () => {
+const LoadNewsButton = ({ showNews, isChecked, toggleChange }) => {
   return (
-    <div id="news-page" className="">
-      <iframe src="https://blog.couchdb.org" width="100%" height="100%"></iframe>
+    <div>
+      <p>
+        When you click this button, you are requesting content and sharing your IP address with <a href="https://blog.couchdb.org/">blog.couchdb.org</a> which is edited by the Apache CouchDB PMC and maintained by <a href="https://wordpress.com/">wordpress.com</a>.
+      </p>
+      <p>
+        If you don’t want to share your IP address, do not click the button.
+      </p>
+      <button className="btn btn-primary" onClick={showNews}>Load News</button>
+      <label className="news-checkbox">
+        <input type="checkbox"
+          checked={isChecked}
+          onChange={toggleChange}
+        />
+        Remember my choice
+      </label>
     </div>
   );
 };
 
+class NewsPage extends React.Component {
+  constructor (props) {
+    super(props);
+    this.showNews = this.showNews.bind(this);
+    this.toggleChange = this.toggleChange.bind(this);
+
+    const allowsIpSharing = !!app.utils.localStorageGet('allow-IP-sharing');
+
+    this.state = {
+      showNews: allowsIpSharing ? true : false,
+      allowsIpSharing
+    };
+  }
+
+  showNews() {
+    this.setState({ showNews: true });
+  }
+
+  toggleChange() {
+    const allowsIpSharing = this.state.allowsIpSharing;
+    this.setState(() => ({ allowsIpSharing: !allowsIpSharing }));
```

Comment:
Can you make this `this.setState({ allowsIpSharing: !allowsIpSharing })` and then this is ready to merge. Nice work.

Reply:
@garrensmith I did it :)